packages/teams-js/src/public/runtime.ts

```diff
@@ -73,7 +73,10 @@ interface IRuntimeV1 extends IBaseRuntime {
       };
     };
     readonly teamsCore?: {};
-    readonly video?: {};
+    readonly video?: {
+      readonly mediaStream?: {};
```

Comment:
@TrevorJoelHarris do you know which host is using the v1 runtime? I can see T1 is using the default `teamsRuntimeConfig` which is actually a v2 runtime, I added a `sharedFrame` subcapability to it, does it mean I don't need to do anything in the `upgradeChain`?

Reply:
v1 runtime will be any hosts that haven't upgrade to use a recent release of the host-sdk that @erinha owns. Whoever is using it, they will be using a version of the host code that doesn't know about this change or that it should be sending down updated runtime objects. Because of this, this section of code handles "what do we do if we get an older version of the runtime object". If we get a version of runtime that is too early to know about this change from a host then runtime.upgradeChain needs to handle correctly setting the right capabilities to maintain correctness across multiple versions of runtime. As I mentioned, @jekloudaMSFT should be able to help you with details if needed.

Reply:
Hi @WengHe I think this is going to require a new version of the runtime object. If I understand correctly, hubs that are currently sending a runtime where `video` is present should send `video.sharedFrame` now, right? If that's the case, then you will need to add a new version of the runtime (it should be 2.1), and then add a function to `upgradeChain` that converts from a version 2 runtime to a version 2.1 runtime by converting `video {}` to `video { sharedFrame {} }`. There's a runtime.md document that explains how to update the runtime version.

Reply:
I wonder a new version of the runtime object is needed or not.

For now, the `video` capability is only supported by Teams (T1 and T2/T2.1). T1 should have `video.sharedFrame` supported while T2.1 should have `video.mediaStream` supported. T2.1 is using the latest hub sdk, T1 doesn't use hub sdk so there is no runtime pass down to the teamsjs sdk, thus there is a default runtime defined in the teamsjs sdk, named `teamsRuntimeConfig`.

Therefore, what I am doing in the PR is:

* update runtimev2, add subcapability to video: _video?: {sharedFrame?:{}, mediaStream?:{}}_
* update ⁠teamsRuntimeConfig, add _video.sharedFrame = {}_

Without a need to update the upgradeChain.

Reply:
Ok, if Teams is the only host using the `video` capability, then that should be fine